gnes/preprocessor/helper.py

```diff
@@ -43,6 +43,64 @@ def get_video_length(video_path):
     return 3600 * h + 60 * m + s
 
 
+def get_video_length_from_raw(buffer_data):
+    import re
+    import datetime
+    ffmpeg_cmd = ['ffmpeg', '-i', '-', '-']
+    with sp.Popen(ffmpeg_cmd, stdin=sp.PIPE, stdout=sp.PIPE, stderr=sp.PIPE,
+                        bufsize=-1, shell=False) as pipe:
+        _, stdout = pipe.communicate(buffer_data)
+        stdout = stdout.decode()
+        matches = re.search(r"Duration:\s{1}(?P<hours>\d+?):(?P<minutes>\d+?):(?P<seconds>\d+\.\d+?),", stdout, re.DOTALL).groupdict()
+        h = str(int(matches['hours']))
+        m = str(int(matches['minutes']))
+        s = str(round(float(matches['seconds'])))
+        duration = datetime.datetime.strptime(h + ':' + m + ':' + s, '%H:%M:%S')
+    return duration
+
+
+def get_audio(buffer_data, sample_rate, interval,
+              duration) -> List['np.ndarray']:
+    import datetime
+    import numpy as np
+    import soundfile as sf
+    from datetime import timedelta
```

Comment:
only 3rd party package needs to be import inside the function.
for python standard library and numpy, there is no need to do that.